ANDROID: Add rt_sigpending local test case

Add a local test case `rt_sigpending02` under `testcases/kernel/syscalls/rt_sigpending`
which builds `rt_sigpending02` from the existing `sigpending02.c` with
-DTEST_RT_SIGPENDING.

Test: mm

diff --git a/testcases/kernel/syscalls/rt_sigpending/.gitignore b/testcases/kernel/syscalls/rt_sigpending/.gitignore
new file mode 100644
index 000000000..8e76ad164
--- /dev/null
+++ b/testcases/kernel/syscalls/rt_sigpending/.gitignore
@@ -0,0 +1 @@
+/rt_sigpending02
diff --git a/testcases/kernel/syscalls/rt_sigpending/Makefile b/testcases/kernel/syscalls/rt_sigpending/Makefile
new file mode 100644
index 000000000..8e76ad164
--- /dev/null
+++ b/testcases/kernel/syscalls/rt_sigpending/Makefile
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2019, Linux Test Project
+
+top_srcdir		?= ../../../..
+
+include $(top_srcdir)/include/mk/testcases.mk
+
+CPPFLAGS += -DTEST_RT_SIGPENDING
+
+rt_sigpending02: $(abs_srcdir)/../sigpending/sigpending02.c
+	$(LINK.c) $^ $(LOADLIBES) $(LDLIBS) $(OUTPUT_OPTION)
+
+MAKE_TARGETS := rt_sigpending02
+
+include $(top_srcdir)/include/mk/generic_leaf_target.mk
